uefi-test-runner/src/boot/memory.rs

```diff
@@ -1,17 +1,46 @@
+use uefi::boot;
 use uefi::table::boot::{AllocateType, BootServices, MemoryMap, MemoryMapMut, MemoryType};
 
 use alloc::vec::Vec;
 
 pub fn test(bt: &BootServices) {
     info!("Testing memory functions");
 
+    test_allocate_pages_freestanding();
+    test_allocate_pool_freestanding();
+
     allocate_pages(bt);
     vec_alloc();
     alloc_alignment();
 
     memory_map(bt);
 }
 
+fn test_allocate_pages_freestanding() {
+    let num_pages = 1;
+    let addr =
+        boot::allocate_pages(AllocateType::AnyPages, MemoryType::LOADER_DATA, num_pages).unwrap();
+    assert_eq!(addr % 4096, 0, "Page pointer is not page-aligned");
+
+    // Verify the page can be written to.
+    {
+        let buf = unsafe { &mut *(addr as *mut [u8; 4096]) };
+        *buf = [0xff; 4096];
```

Comment:
I think we should use core::ptr::write_volatile so that this is not optimized away

Reply:
Good catch, updated.